scripts/docker.sh

```diff
@@ -58,26 +59,28 @@ function docker_build {
   build_args+=(--build-arg "BUILDER_IMAGE=${BUILDER_IMAGE}")
   build_args+=(--build-arg "BASE_IMAGE=${BASE_IMAGE}")
 
-  # Use + conditional parameter expansion to protect from unbound array variable
-  docker build \
-    -t "${GCR_REGISTRY}/${name}:${UNSTABLE_TAG}" \
-    -f "${dockerfile}" \
-    "${build_args[@]+"${build_args[@]}"}" \
-    "${function_dir}"
-}
-
-function docker_push {
-  name=$1 # function name, e.g. apply-setters
-  version=$2 # function version, e.g. v0.1.1
-
-  docker push "${GCR_REGISTRY}/${name}:${version}"
-}
-
+  echo "building ${GCR_REGISTRY}/${name}:${tag}"
 
-function docker_tag {
-  name=$1 # function name, e.g. apply-setters
-  version=$2 # function version, e.g. v0.1.1
-
-  echo tagging "${GCR_REGISTRY}/${name}:${version}"
-  docker tag "${GCR_REGISTRY}/${name}:${UNSTABLE_TAG}" "${GCR_REGISTRY}/${name}:${version}"
+  case "${action}" in
+    load)
+      # Use + conditional parameter expansion to protect from unbound array variable
+      docker buildx build --load \
+        -t "${GCR_REGISTRY}/${name}:${tag}" \
+        -f "${dockerfile}" \
+        "${build_args[@]+"${build_args[@]}"}" \
+        "${function_dir}"    
+      ;;
+    push)
+      # build and push multi-arch image.
+      docker buildx build --push \
+        -t "${GCR_REGISTRY}/${name}:${tag}" \
+        -f "${dockerfile}" \
+        --platform "linux/amd64,linux/arm64" \
```

Comment:
Build multi-arch images when pushing to a registry.

Reply:
Will it work for `OSX` or just `linux` ?

Reply:
It will work on mac.
`container` is a linux-only thing. On mac, containers are executed in a linux VM.

ARM has v7 and v8. From my understanding, Apple's chips are all v8. The default is v8 when we are using linux/arm64.
We can add v7 in the future if needed.

Reply:
Cool.